full-stack/orders-filters-and-listing.md

```diff
@@ -0,0 +1,128 @@
+  instrumentId: { type: String, required: True },
+
+  purchaseType: { type: String, required: True, enum: [ 'BUY', 'SELL' ] },
+  countparty: {
+    name: { type: String, required: True },
+  },
+
+  dataSource: {
+    name: { type: String, required: True },
+  },
+
+  tradeData: {
+    quantity: { type: String, required: True },
+    price: { type: Number, required: True },
+    trader: { type: String, required: True },
+  },
+
+  timestamps: {
+    orderSubmitted: { type: ISODateString, required: True },
+  }
+}
+```
+
+## Database 
+```
+DATABSE_URL=<INSERT_HERE>
+INDEX_NAME=.order
+```
+
+## API:
+Write an API using Python 
+
+```
+GET /listings?type=BUY&search=Amazon&assetClass=Equity&minOrderSubmitted=2011-10-05T14:48:00.000Z&maxOrderSubmitted=2020-12-05T14:48:00.000Z&minPrice=120&maxPrice=2000&minQuantity=10&axQuantity=2000
+```
+
+
+| Parameter           | Description                                                                  |
+|---------------------|------------------------------------------------------------------------------|
+| `assetClass`        | Asset class of the order.                                                    |
+| `maxOrderSubmitted` | The maximum date for `timestamps.orderSubmitted` field.                      |
+| `maxPrice`          | The maximum value for `tradeData.price` field.                               |
+| `minOrderSubmitted` | The minimum date for `timestamps.orderSubmitted` field.                      |
+| `minPrice`          | The minimum value for `tradeData.price` field.                               |
+| `search`            | Search on the `instrumentName`, `instrumentId` and `counterparty.name` field |
+| `purchaseType`      | The trade is BUY or SELL.                                                    |
+| `limit`             | Maximum number of orders to send at once., defaults to 10                    |
+| `skip`              | The number of orders to skip, defaults to 0                                  |
+
+
+The expected response is a paginated JSON orders of listings:
+
+```
+{
+  header: {
+    totalHits: 30,
+  },
+  results: [
+    {
+      assetClass: 'Equity',
+      countpartyName: 'John Smith',
+      dataSourceName: 'SteelEye',
+      instrumentId: 'AMZN',
+      instrumentName: 'AMAZON NYE',
+      orderId: 'ORD-343243',
+      orderSubmitted: '2020-10-05T14:48:00.000Z',
+      purchaseType: 'BUY',
+      tradePrice: 10000,
+      tradeQuantity: 100,
+    }
+    ...
+  ]
+}
+```
+
+All query parameters are optional, all minimum and maximum fields should be inclusive (e.g. minPrice=2&maxPrice=400 should return orders with 2, 3, or 400 trade price).
+
+At a minimum:
+- Your API endpoint URL is `/listings`
+- Your API should correctly filter any combination of API parameters
+
+## Frontend
+
+Create a simple web page using Angular / React or any other framework of choice for the the above filters listing. You can use any library for date input and price range input if needed.
+
```

Comment:
Why Angular? If we're expecting this person to write React code, the test submissions should be in React